text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1601 @@
+
+#### From Other Signals to Profiles
+
+Other signals can use `profile_id` to reference a profile. For example, a log record can reference a profile that was collected at the time when the log record was generated by using `profile_id` as one of the attributes. This allows to correlate logs with profiles.
+
+Additionally, `trace_id`, `span_id` can be used to reference specific [Samples](#message-sample) in a Profile, since [Samples](#message-sample) are linked to traces with these same identifiers using [Links](#message-link).
+
+The exact details of such linking are out of scope for this OTEP. It is expected that the exact details will be defined in Profiles part of [opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification).
+
+### Compatibility With Original pprof
+
+The proposed data model is backwards compatible with original pprof in a sense that a pprof file generated by existing software can be parsed using the new proto.
+
+It is not forwards compatible, meaning that a pprof file generated by the new proto cannot be parsed by existing software. This is mainly due to the sharing of the call stacks between samples + new format for labels (more on these differences below).
+
+### Proto Definition
+
+Proto definition is based on [pprof format](https://github.com/google/pprof/blob/main/proto/profile.proto).
+
+In the landscape of performance profiling tools, pprof's data format stands as a clear industry standard. Its evolution and enduring relevance are a reflection of its effectiveness in addressing diverse and complex performance profiling needs. Major technology firms and open-source projects alike routinely employ pprof, underscoring its universal applicability and reliability.
+
+According to the [data from Profilerpedia](https://docs.google.com/spreadsheets/d/1UM-WFQhNf4GcyXmluSUGnMbOenvN-TqP2HQC9-Y50Lc/edit?usp=sharing), it's one of the most widely used formats. Compared to other formats it has the highest number of profilers, UIs, formats it can be converted to and from.
+
+The original pprof data model underwent enhancements to more effectively manage profiling data within the scope of OpenTelemetry, and certain upgrades were implemented to overcome a few of the original format's constraints.
+
+Here's a [link to a diff between original pprof and modified pprof](https://github.com/open-telemetry/opentelemetry-proto-profile/compare/2cf711b3cfcc1edd4e46f9b82d19d016d6d0aa2a...petethepig:opentelemetry-proto:pprof-experiments#diff-9cb689ea05ecfd2edffc39869eca3282a3f2f45a8e1aa21624b452fa5362d1d2) and here's a list of main differences between pprof and OTLP profiles:
+
+* Sharing of the call stacks between samples.
+* Sharing of labels (now called attributes) between samples.
+* Reuse of OpenTelemetry conventions and message types.
+* Semantic conventions for linking to other signals via `trace_id`s and `span_id`s.
+* First-class timestamp support.
```

Comment:
Remove? We postponed this.

Reply:
I addressed this in [Open Questions section](https://github.com/petethepig/oteps/blob/1e71ed0aa8d16c241ff292da48a44bf36988773f/text/profiles/0239-profiles-data-model.md#timestamps) — my idea was to have some field that we could start using in experiments but make it clear that it will most likely change by the time profiles reach `Stable` state.

I'm following similar logic to that in the [Aggregation Temporality discussion](https://github.com/open-telemetry/oteps/pull/239#discussion_r1462653940) — this gives us a good base to experiment with, while preserving optionality.

I'm open to change this though, let's further discuss Thursday.